pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -308,3 +334,62 @@ func createValidImageVersionLabel(image string) string {
 	}
 	return cephImageVersion
 }
+
+func waitForPvcToExpandWithTimeout(ctx context.Context, client client.Client, pvcResizeMap map[string]pvcResize, namespace string, pvcResizeWaitTimeOut time.Duration) {
+	currentTime := time.Now()
+	if pvcResizeWaitTimeOut == 0 {
+		pvcResizeWaitTimeOut = defaultWaitTimeoutForHealthyOSD
+	}
+	timeout := currentTime.Add(pvcResizeWaitTimeOut)
+	allPvcResize := false
+	for !allPvcResize {
```

Comment:
nit: IMO it's more readable just to have an infinite for loop, and return when there is a timeout or they are all resized, but maybe it's just a personal preference.
```suggestion
	for {
```